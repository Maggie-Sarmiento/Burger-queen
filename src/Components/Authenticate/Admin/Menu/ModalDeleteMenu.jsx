/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import Modal from '@mui/material/Modal';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

const modal = {
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'background.paper',
  border: '1px solid #000',
  borderRadius: 1,
  boxShadow: 24,
  p: 4,
  textAlign: 'center',
};

const ModalDeleteMenu = ({
  menu, product, openModalDelete, setOpenModalDelete,
}) => {
  const [dataMenu, setDataMenu] = useState([]);
  const data = product;

  const handleClose = () => setOpenModalDelete(false);

  const deleteDataApi = (id) => {
    const requestOption = {
      method: 'DELETE',
    };
    if (menu === 'menu') {
      fetch(`http://localhost:3001/menu/${id}`, requestOption)
        .then((response) => response.json())
        .then(() => {
          setDataMenu(dataMenu.filter((itemMenu) => itemMenu.id !== data.id));
        })
        .catch((err) => console.log(err));
    } else {
      fetch(`http://localhost:3001/menuLunch/${id}`, requestOption)
        .then((response) => response.json())
        .then(() => {
          setDataMenu(dataMenu.filter((itemMenu) => itemMenu.id !== data.id));
        })
        .catch((err) => console.log(err));
    }

    handleClose();
  };

  return (
    <Modal
      open={openModalDelete}
      onClose={handleClose}
    >
      <Box sx={modal}>
        <div>
          <p>
            Estás seguro que deseas eliminar a
            <br />
            <span>{data && data.name}</span>
            ?
          </p>
          <div align="right">
            <Button color="secondary" onClick={() => deleteDataApi(data.id)}>Sí</Button>
            <Button onClick={() => handleClose()}>No</Button>
          </div>
        </div>
      </Box>
    </Modal>
  );
};

export default ModalDeleteMenu;
