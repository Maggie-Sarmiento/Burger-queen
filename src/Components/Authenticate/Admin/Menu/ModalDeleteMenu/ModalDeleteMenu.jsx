/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import Modal from '@mui/material/Modal';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import { urlServer } from '../../../../../config';

const modal = {
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'background.paper',
  border: '1px solid #000',
  borderRadius: 1,
  boxShadow: 24,
  p: 4,
  textAlign: 'center',
};

const ModalDeleteMenu = ({
  menu, product, openModalDelete, setOpenModalDelete, refreshData, setRefreshData,
}) => {
  const [dataMenu, setDataMenu] = useState([]);
  const data = product;

  const handleClose = () => {
    setOpenModalDelete(false);
  };

  const refresh = () => setRefreshData(!refreshData);

  const deleteDataApi = (id) => {
    const requestOption = {
      method: 'DELETE',
      headers: { 'Content-Type': 'application/json', Authorization: localStorage.getItem('token') },
    };
    if (menu === 'menu') {
      fetch(`${urlServer}/menu/${id}`, requestOption)
        .then((response) => {
          response.json();
          refresh();
        })
        .then(() => {
          setDataMenu(dataMenu.filter((itemMenu) => itemMenu.id !== data.id));
        })
        .catch((err) => console.log(err));
    } else {
      fetch(`${urlServer}/menuLunch/${id}`, requestOption)
        .then((response) => {
          response.json();
          refresh();
        })
        .then(() => {
          setDataMenu(dataMenu.filter((itemMenu) => itemMenu.id !== data.id));
        })
        .catch((err) => console.log(err));
    }

    handleClose();
  };

  return (
    <Modal
      open={openModalDelete}
      onClose={handleClose}
    >
      <Box sx={modal}>
        <div>
          <p>
            Estás seguro que deseas eliminar a
            <br />
            <span>{data && data.name}</span>
            ?
          </p>
          <div align="right">
            <Button color="secondary" onClick={() => deleteDataApi(data._id)}>Sí</Button>
            <Button onClick={handleClose}>No</Button>
          </div>
        </div>
      </Box>
    </Modal>
  );
};

export default ModalDeleteMenu;
