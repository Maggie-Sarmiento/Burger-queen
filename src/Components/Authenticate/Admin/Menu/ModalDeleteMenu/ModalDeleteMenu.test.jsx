/* eslint-disable react/jsx-boolean-value */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import ModalDeleteMenu from './ModalDeleteMenu';

describe('test ModalDeleteMenu', () => {
  test('render component ModalDeleteMenu', async () => {
    let root;
    act(() => {
      root = create(
        <Router>
          <ModalDeleteMenu />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
  test('Click button', async () => {
    const mockOnClose = jest.fn();
    render(
      <Router>
        <ModalDeleteMenu
          openModalDelete={true}
          setOpenModalDelete={mockOnClose}
        />
      </Router>,
    );
    const button = screen.getByText(/No/i);
    fireEvent.click(button);
    expect(button).toBeInTheDocument();
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });
});
