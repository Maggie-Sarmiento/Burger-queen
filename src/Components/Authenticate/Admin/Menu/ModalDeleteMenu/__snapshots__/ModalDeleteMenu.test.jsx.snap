// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ModalDeleteMenu render component ModalDeleteMenu 1`] = `null`;
